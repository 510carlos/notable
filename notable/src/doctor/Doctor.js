import React, { useEffect, useState } from 'react';
import { getApointment } from './Doctors.helpers';
import { Apointments, Information } from './components';

const Doctor = ({match}) => {
    const id = match.params.id || -1;

    const [apt, setApt] = useState([]);

    useEffect( () => {
        let mounted = true;
        getApointment(id).then( result => {
        const { data } = result;
        if (mounted)
            setApt(data)
        });
        return () => {
            mounted = false;
        }
    }, [id]);

    return (
        <>
            <Information id={id} />
            <Apointments apt={apt} />
        </>
    )
}

export default Doctor;