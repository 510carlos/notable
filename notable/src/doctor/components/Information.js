import React from 'react';
import { useDoctorContext } from '../Doctor.context';


const Information = ({id}) => {
    const doctorsList = useDoctorContext();
    let doctor = {
        "name": "n/a",
        "email": "n/a"
    };
    if(doctorsList && doctorsList[id])
        doctor = doctorsList[id];

    return (
        <>
            <h1>{doctor.name}</h1>
            <h3>{doctor.email}</h3>
        </>
    )
};

export default Information;