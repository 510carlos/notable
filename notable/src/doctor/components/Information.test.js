import * as React from 'react';

import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import Information from './Information';

import * as DoctorContext from '../Doctor.context';

const docId = 1;
const dom = <Information id="1" />;

const contextValues = [{"id":1,"name":"Hibbert, Julius","email":"kriger@notablehealth.com"},{"id":2,"name":"Krieger, Algemap","email":"kriger@notablehealth.com"},{"id":3,"name":"Riviera, Nick","email":"kriger@notablehealth.com"}];
const doc = contextValues[docId];

jest.spyOn(DoctorContext, 'useDoctorContext')
    .mockImplementation(() => contextValues);

describe('Unit tests for Information', () => {
    it('Renders', () => {
        shallow(dom)
    })

    it('Renders with snapshot', () => {
        const wrapper = shallow(dom);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    })

    it('It renders with the correct props', () => {
        const wrapper = shallow(dom);
        const name = wrapper.find('h1').text();
        const email = wrapper.find('h3').text();
        expect(doc.name).toEqual(name);
        expect(doc.email).toEqual(email);
    })
})