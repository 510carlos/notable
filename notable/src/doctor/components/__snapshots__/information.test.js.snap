// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for Information Renders with snapshot 1`] = `
<Fragment>
  <h1>
    Krieger, Algemap
  </h1>
  <h3>
    kriger@notablehealth.com
  </h3>
</Fragment>
`;
